web/src/components/HeaderMenu.tsx

```diff
@@ -0,0 +1,139 @@
+import CheckIcon from '@mui/icons-material/Check'
+import FacebookOutlinedIcon from '@mui/icons-material/FacebookOutlined'
+import LinkIcon from '@mui/icons-material/Link'
+import MailOutlinedIcon from '@mui/icons-material/MailOutlined'
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import ShareIcon from '@mui/icons-material/ShareOutlined'
+import WhatsAppIcon from '@mui/icons-material/WhatsApp'
+import { dividerClasses } from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import MuiMenu from '@mui/material/Menu'
+import { styled } from '@mui/material/styles'
+import React, { ReactElement, RefObject, useImperativeHandle } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import buildConfig from '../constants/buildConfig'
+import { useRouteParams } from '../hooks/useCityContentParams'
+import useDimensions from '../hooks/useDimensions'
+import { withDividers } from '../utils'
+import getFooterLinks from '../utils/getFooterLinks'
+import MenuAccordion from './MenuAccordion'
+import MenuItem from './MenuItem'
+
+const COPY_TIMEOUT = 3000
+
+const StyledMenu = styled(MuiMenu)({
+  marginTop: 8,
+
+  [`& .${dividerClasses.root}`]: {
+    margin: '0 !important',
+  },
+})
+
+export type MenuRef = {
+  closeMenu: () => void
+}
+
+type HeaderMenuProps = {
+  children: (ReactElement | null)[] | ReactElement
+  pageTitle: string | null
+  ref?: RefObject<MenuRef | null>
+}
+
+const HeaderMenu = ({ children, pageTitle, ref }: HeaderMenuProps): ReactElement => {
+  const [menuAnchorElement, setMenuAnchorElement] = React.useState<HTMLElement | null>(null)
+  const [expandedAccordion, setExpandedAccordion] = React.useState<'share' | 'legal' | null>(null)
+  const [urlCopied, setUrlCopied] = React.useState<boolean>(false)
+  const { cityCode, languageCode } = useRouteParams()
+  const { mobile } = useDimensions()
+  const { t } = useTranslation('layout')
+
+  useImperativeHandle(ref, () => ({ closeMenu: () => setMenuAnchorElement(null) }))
+
+  const openMenu = (event: React.MouseEvent<HTMLElement>) => setMenuAnchorElement(event.currentTarget)
+  const closeMenu = () => setMenuAnchorElement(null)
+  const open = menuAnchorElement !== null
+
+  const items = Array.isArray(children) ? children : [children]
+
+  const shareUrl = window.location.href
+  const encodedTitle = encodeURIComponent(pageTitle ?? buildConfig().appName)
+  const encodedShareUrl = encodeURIComponent(shareUrl)
```

Comment:
```suggestion
  const encodedShareUrl = encodeURIComponent(shareUrl)
  const encodedTitle = encodeURIComponent(pageTitle ?? buildConfig().appName)
```
Just to keep the things that refer to each other together